### PR TITLE
Bump golang, dependencies, replace golint with staticcheck

### PR DESCRIPTION
**Pull request by antoinedeschenes**
- Bump golang to 1.20 (syncing with helm)
- Bump helm dependency to 3.13
- Bump k8s libraries to 0.28.4 (syncing with helm)
- Replace deprecated golint linter with staticcheck
- Fix deprecations

### CODE REVIEW

File: staticcheck.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+checks = ["all", "-ST1000", "-ST1005"]
```
Comment (antoinedeschenes) [inline staticcheck.conf:1]: -ST1000: ignore missing package comments
-ST1005: ignore capitalized error messages

File: scripts/verify-staticcheck.sh [hunk 2]
```diff
@@ -1,20 +1,19 @@
 #!/bin/bash
 # script credits : https://github.com/infracloudio/botkube
 
-set -o errexit
```
Comment (antoinedeschenes) [inline scripts/verify-staticcheck.sh:4]: This would exit early before the error messages were echoed out.

**Review by yxxhero — APPROVED** [review]

**Comment by yxxhero** [comment]
LGTM

**Comment by yxxhero** [comment]
@mumoshu WDYT?

File: .circleci/config.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: cimg/go:1.19
+      - image: cimg/go:1.21
```
Comment (mumoshu) [inline .circleci/config.yml:5]: Thanks! But we've already stopped using circleci so you could just remove this out of this pull request!

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 name: CI
 
 env:
-  VERSION_GO: '1.19.8'
+  VERSION_GO: '1.21.5'
```
Comment (mumoshu) [inline .github/workflows/ci.yaml:5]: We'd better keep it aligned with the latest helm version that helm-diff supports! It should be the latest of 1.20 this time then?

https://github.com/helm/helm/blob/c8b948945e52abba22ff885446a1486cb5fd3474/.github/workflows/release.yml#L28

File: staticcheck.conf (hunk 1 above)
Comment (mumoshu) [reply staticcheck.conf:1]: Thanks! This looks like great for the initial list!

We'd better address ST1005 and remove "-ST1005" in due course.

https://staticcheck.dev/docs/checks/#ST1005

File: Dockerfile.release [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19.8
+FROM golang:1.21.5
```
Comment (mumoshu) [inline Dockerfile.release:1]: https://github.com/databus23/helm-diff/pull/510/files#r1429342804

File: go.mod [hunk 6]
```diff
@@ -1,6 +1,6 @@
 module github.com/databus23/helm-diff/v3
 
-go 1.19
+go 1.21
```
Comment (mumoshu) [inline go.mod:3]: https://github.com/databus23/helm-diff/pull/510/files#r1429342804

File: go.mod [hunk 7]
```diff
@@ -11,73 +11,73 @@ require (
...
-	k8s.io/apiextensions-apiserver v0.26.0
-	k8s.io/apimachinery v0.26.2
-	k8s.io/cli-runtime v0.26.0
-	k8s.io/client-go v0.26.2
+	helm.sh/helm/v3 v3.13.3
+	k8s.io/api v0.29.0
+	k8s.io/apiextensions-apiserver v0.29.0
+	k8s.io/apimachinery v0.29.0
+	k8s.io/cli-runtime v0.29.0
+	k8s.io/client-go v0.29.0
```
Comment (mumoshu) [inline go.mod:21]: You'd need to stick with v0.28.x for helm 3.13.3
https://github.com/helm/helm/blob/c8b948945e52abba22ff885446a1486cb5fd3474/go.mod#L41-L44

File: .circleci/config.yml (hunk 3 above)
Comment (antoinedeschenes) [reply .circleci/config.yml]: File deleted

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (antoinedeschenes) [reply .github/workflows/ci.yaml]: switched to 1.20.12

File: go.mod [hunk 8]
```diff
@@ -11,73 +11,73 @@ require (
...
-	google.golang.org/grpc v1.53.0 // indirect
+	github.com/stretchr/testify v1.8.4
 	gopkg.in/yaml.v2 v2.4.0
-	helm.sh/helm/v3 v3.11.3
-	k8s.io/api v0.26.2
-	k8s.io/apiextensions-apiserver v0.26.0
-	k8s.io/apimachinery v0.26.2
-	k8s.io/cli-runtime v0.26.0
-	k8s.io/client-go v0.26.2
+	helm.sh/helm/v3 v3.13.3
```
[the remark this answers is not in the document]
Comment (antoinedeschenes) [reply go.mod:18]: done

File: go.mod [hunk 9]
```diff
@@ -1,6 +1,6 @@
 module github.com/databus23/helm-diff/v3
 
-go 1.21
+go 1.20
```
Comment (antoinedeschenes) [inline go.mod:3]: That one is a downgrade, syncing go version to helm.

**Review by mumoshu — APPROVED** [review]
LGTM! Approved to trigger the CI build. Let's merge once it passes.

Thank you very much for your contribution @antoinedeschenes (and review @yxxhero) ☺️

**Review by yxxhero — APPROVED** [review]

**Comment by yxxhero** [comment]
@mumoshu all is pass.

**Comment by antoinedeschenes** [comment]
Yup the `CircleCI Pipeline` webhook would need to be removed I assume

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks for your efforts!

**Pull request merged**